### ISSUE TITLE
Shoots stuck in reconcile because of certificate-service extension

### ISSUE THREAD

**Issue by schrodit** [gardener/gardener#1526]
**What happened**:
```
retry failed with context deadline exceeded, last error: extension certificate-service is still present
```

Shoot clusters stuck in reconcile because the `certificate-service` extension cannot be deleted correctly, although a deletion timestamp is set correctly.
When retrigger deletion with `kubectl delete extensions.extensions.gardener.cloud certificate-service` it is deleted immediately

Example:
```
apiVersion: extensions.gardener.cloud/v1alpha1
kind: Extension
metadata:
  creationTimestamp: "2019-09-15T07:50:07Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2019-10-15T21:03:29Z"
  finalizers:
  - foregroundDeletion
  generation: 2
  name: certificate-service
  namespace: <shoot-name>
  resourceVersion: "481927309"
  selfLink: /apis/extensions.gardener.cloud/v1alpha1/namespaces/<shoot-name>/extensions/certificate-service
  uid: <uid>
spec:
  type: certificate-service
status:
  lastOperation:
    description: Successfully deleted Extension resource
    lastUpdateTime: "2019-10-22T06:04:06Z"
    progress: 100
    state: Succeeded
    type: Delete
  observedGeneration: 2
```

**What you expected to happen**:
The extension should be deleted and the shoot should finish reconcile.

**Environment**:

- Gardener version: 0.30.5
- Kubernetes version (use `kubectl version`): Seed 1.15.5
- Cloud provider or hardware configuration: AWS

**Comment by rfranzke** [comment]
Root cause is a bug in Kubernetes, see https://github.com/kubernetes/kubernetes/issues/77081

**Comment by schrodit** [comment]
Ok so fixed in 1.16. Then I will just delete all extensions manually 👍
